### PR TITLE
Add support for ppro survey

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/1201462763415876/1207541002825789/f

### Description
See attached task description

### Steps to test this PR
https://app.asana.com/0/1201462763415876/1207598869934445/f

### CODE REVIEW

**Comment by karlenDimla** [comment]
* **#4661** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4661?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karlenDimla and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: survey/survey-api/src/main/java/com/duckduckgo/survey/api/SurveyParameterManager.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.survey.api
+
+interface SurveyParameterManager {
+    suspend fun canProvideAllParameters(requestedQueryParams: List<String>): Boolean
```
Comment (aitorvs) [inline survey/survey-api/src/main/java/com/duckduckgo/survey/api/SurveyParameterManager.kt:20]: Why do we need this method?
I'd also add javadocs to public api methods.

File: survey/survey-impl/src/main/java/com/duckduckgo/survey/impl/SurveyParameterProviderPluginPoint.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import com.duckduckgo.common.utils.plugins.PluginPoint
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.survey.api.SurveyParameterPlugin
+
+@ContributesPluginPoint(
+    scope = AppScope::class,
+    boundType = SurveyParameterPlugin::class,
+)
+@Suppress("unused")
+interface SurveyParameterProviderPluginPoint
```
Comment (aitorvs) [inline survey/survey-impl/src/main/java/com/duckduckgo/survey/impl/SurveyParameterProviderPluginPoint.kt:29]: This should be private

File: survey/survey-api/src/main/java/com/duckduckgo/survey/api/SurveyParameterManager.kt (hunk 1 above)
Comment (karlenDimla) [reply survey/survey-api/src/main/java/com/duckduckgo/survey/api/SurveyParameterManager.kt]: Updated!

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/rmf/RMFPProDaysSinceSubscribedMatchingAttribute.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+class RMFPProDaysSinceSubscribedMatchingAttribute @Inject constructor(
+    private val subscriptionsManager: SubscriptionsManager,
+    private val currentTimeProvider: CurrentTimeProvider,
+) : JsonToMatchingAttributeMapper, AttributeMatcherPlugin {
+    override suspend fun evaluate(matchingAttribute: MatchingAttribute): Boolean? {
+        return when (matchingAttribute) {
+            is PProDaysSinceSubscribedMatchingAttribute -> {
+                val subscription = subscriptionsManager.getSubscription()
+                return if (subscription == null || (
+                    matchingAttribute.min.isDefaultValue() &&
```
Comment (cmonfortep) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/rmf/RMFPProDaysSinceSubscribedMatchingAttribute.kt:49]: you could compare `matchingAttribute` against `emptyValuesAttribute`. And just create emptyValues as PProDaysSinceSubscribedMatchingAttribute()

File: subscriptions/subscriptions-impl/src/test/java/com/duckduckgo/subscriptions/impl/rmf/RMFPProDaysSinceSubscribedMatchingAttributeTest.kt [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        assertEquals(PProDaysSinceSubscribedMatchingAttribute(min = 20, max = 30), result)
+    }
+
+    @Test
+    fun whenMapKeyIsNotPProDaysSinceSubscribedTheMapReturnNull() = runTest {
+        val jsonMatchingAttribute = JsonMatchingAttribute(min = 20, max = 30)
+        val result = matcher.map("somethingelse", jsonMatchingAttribute)
+        assertNull(result)
+    }
+
```
Comment (cmonfortep) [inline subscriptions/subscriptions-impl/src/test/java/com/duckduckgo/subscriptions/impl/rmf/RMFPProDaysSinceSubscribedMatchingAttributeTest.kt:58]: To add a test for the invalid case where we receive value with a non Int type. It should return null or exception.

**Review by cmonfortep — APPROVED** [review]
Looks great @karlenDimla 

I will need help testing this but code changes looks good to me, just minor suggestions. Approving since I'm AFK on Monday and don't want to block you here.

If we have some time, maybe we can also move some stuff from surveys into the new module on another PR, so we can clean that part.

File: survey/survey-impl/src/test/java/com/duckduckgo/survey/impl/RealSurveyParameterManagerTest.kt [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        val result = surveyParameterManager.buildSurveyUrlStrict("http://example.com", listOf("hello", "ppro"))
+
+        assertNull(result)
+    }
+
+    @Test
+    fun whenNoRequestedParamsThenBuildSurveyUrlStrictReturnsResolvedUrl() = runTest {
+        val result = surveyParameterManager.buildSurveyUrlStrict("http://example.com", emptyList())
+
+        assertNotNull(result)
```
[the remark this answers is not in the document]
Comment (karlenDimla) [reply survey/survey-impl/src/test/java/com/duckduckgo/survey/impl/RealSurveyParameterManagerTest.kt:70]: Ah good point. This was a result of changes in API. Fixed!

**Comment by karlenDimla** [comment]
Tested by @anikiki 👍

**Pull request merged**